core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -87,6 +98,18 @@ public static MetricsConfig fromProperties(Map<String, String> props) {
       spec.defaultMode = MetricsModes.fromString(DEFAULT_WRITE_METRICS_MODE_DEFAULT);
     }
 
+    // First set sorted column with sorted column default (can be overridden by user)
+    final Set<String> sortedCols = new HashSet<>();
```

Comment:
nit: we usually don't mark temp vars as final in Iceberg. It is a style preference. I know there were some performance optimizations in old JVMs for final vars but I think modern JVM are smart enough these days.

Reply:
You could probably init the set to the method call result instead of calling calling `addAll` later.

```
Set<String> sortColumns = order != null ? SortOrderUtil.sortColumns(order) : ImmutableSet.of();
...
```

Reply:
Or something like [this](https://github.com/apache/iceberg/pull/2240#discussion_r684560856). Up to you.

Reply:
Yea true I dont usually put final on local variables, don't recall how that happened.  Anyway, simplified according to your suggestion.